csrc/preseg_passes/allocation_order_inference.cpp

```diff
@@ -95,6 +96,12 @@ class AllocationOrderInferencer : public IterVisitor {
       const std::vector<IterDomain*>& permutation_ref) {
```

Comment:
This is an overload function.
`permutation_ref` is used to compute the allocation order for consumer.

1. construct alloc_domain_producer for producer given its allocation_order.
2. map the alloc_domain_producer to consumer as alloc_domain.
3. compute permutation from `alloc_domain` to `permutation_ref` and store that as the allocation_order for consumer.

Usually we call `propagateAllocationOrder(producer, consumer)`, we'll by default use `consumer->getMaybeRFactorDomain()` as `permutation_ref`.

But this won't work for `pad`. since its output rfactor_domain is a resize and won't map to its producer any more. Hence we have this overload and would use its root_domain as reference instead: https://github.com/NVIDIA/Fuser/blob/79862aec572a9149fda7e464c2cc74dbd59bdb79/csrc/preseg_passes/allocation_order_inference.cpp#L295

Reply:
^^^ I think this is another inline comment I should have added. 📓

Reply:
As discussed offline, I think I understand the problem now: while `computePermutation` at line 159 needs rfactor IDs, `propagateAllocationDomain` returns root IDs not necessarily rfactor IDs. So, as a workaround for ops that transform root to rfactor (e.g. pad), we construct `permutation_ref` by hand so that it contains rfactor IDs. 

When we want to handle ViewOp (I guess soon?), we may need to compute permutation_ref  in a more sophisticated way like https://github.com/NVIDIA/Fuser/blob/main/csrc/alias_analysis.cpp#L163. 

SGTM if that's your understanding as well!